app/models/application_record.rb

```diff
@@ -19,6 +19,8 @@ def append_url_options(path, options = {})
       path = "#{path}?#{query_params.to_query}"
     end
 
+    path = "#{path}?preview=#{options[:preview]}" if options[:preview]
```

Comment:
This won't work if both `cachebust` and `preview` are present together as you'll end up with `path?cachebust=1234?preview=1234` instead of `path?cachebust=1234&preview=1234`.  The value for `preview` will need to be added into `query_params` further up.

Perhaps:

```ruby
    query_params = {}
    query_params[:cachebust] = options[:cachebust] if options[:cachebust]
    query_params[:preview] = options[:preview] if options[:preview]

    path = "#{path}?#{query_params.to_query}"
```

A test scenario where both `cachebust` and `preview` are both used at the same time would've caught this.

Reply:
Also sorted, cheers for picking this up, didn't think about both being applied at the same time 👍

Reply:
Got there in the end 👍